### PR TITLE
fix(HLS): Add subtitle role when there are no roles

### LINKED ISSUE #5355: Chrome + HLS subtitles not visible.
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**

**What version of Shaka Player are you using?**
3.3.x

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `main`?**
Yes

**Are you using the demo app or your own custom app?**
Custom App

**If custom app, can you reproduce the issue using our demo app?**
Yes

**What browser and OS are you using?**
Mac 13.3.1 Chromr

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**

**What are the manifest and license server URIs?**

**What configuration are you using?  What is the output of `player.getConfiguration()`?**

**What did you do?**

1. Try to run any Chrome + Hls content on old movies it doesn't show subtitles

**What did you expect to happen?**
1. Subtitles should be shown

**What actually happened?**

hotstarissue

### PR DESCRIPTION
**Pull request by officialbidisha**
Fixes #5355

### CODE REVIEW

**Comment by avelad** [comment]
Can you add some test? Thanks!

**Comment by avelad** [comment]
@officialbidisha please, sign the CLA. Thanks!

**Comment by officialbidisha** [comment]
Already raised a request. Waiting.

**Comment by avelad** [comment]
> Already raised a request. Waiting.

CLA is signed now.

**Comment by officialbidisha** [comment]
Thank you

**Comment by avelad** [comment]
@officialbidisha Please, fix the lint errors and add some test. Thanks!

**Comment by officialbidisha** [comment]
@avelad , please help in fixing the lint errors, it's awaiting approval.

**Comment by officialbidisha** [comment]
Also please help in providing correct PR title @avelad

File: lib/hls/hls_parser.js [hunk 1]
```diff
@@ -2411,6 +2411,20 @@ shaka.hls.HlsParser = class {
...
+      for (const characteristic of characteristics.split(',')) {
+        roles.push(characteristic);
+      }
+    }
+
+    /**
+     * For running chrome in hls
+     */
+    if (!roles.length &&
+      kind === shaka.util.ManifestParserUtils.TextStreamKind.SUBTITLE) {
```
Comment (joeyparrish) [inline lib/hls/hls_parser.js:2425]: > /home/runner/work/shaka-player/shaka-player/lib/hls/hls_parser.js:2425:6: ERROR - [JSC_UNDEFINED_VARIABLE] variable kind is undeclared
>   2425|       kind === shaka.util.ManifestParserUtils.TextStreamKind.SUBTITLE) {

There is no such variable "kind".  You may be thinking of the "kind" field in the anonymous object below.  You should pull that out into a variable first, and reference the variable below in the anonymous object.

**Review by joeyparrish — CHANGES_REQUESTED** [review]

File: test/hls/hls_parser_unit.js [hunk 2]
```diff
@@ -1160,6 +1160,72 @@ describe('HlsParser', () => {
     await testHlsParser(master, media, manifest);
   });
 
+  it('when characteristics are empty then add \n',
```
Comment (joeyparrish) [inline test/hls/hls_parser_unit.js:1163]: This doesn't look right.  it() doesn't take multiple string arguments.  If you need to continue the string on the next line, please do something like this:

```js
it('first part of the line ' +
    'second part of the line', async () => {
```

**Review by joeyparrish — CHANGES_REQUESTED** [review]

File: test/hls/hls_parser_unit.js (hunk 2 above)
Comment (officialbidisha) [reply test/hls/hls_parser_unit.js:1163]: Acknowledged. Will resolve soon.

File: lib/hls/hls_parser.js (hunk 1 above)
Comment (officialbidisha) [reply lib/hls/hls_parser.js:2425]: Agreed, let me make a fix.

Comment (joeyparrish) [reply lib/hls/hls_parser.js]: I edited this and fixed it myself.

File: test/hls/hls_parser_unit.js (hunk 2 above)
Comment (joeyparrish) [reply test/hls/hls_parser_unit.js]: Edited and fixed myself.

File: test/hls/hls_parser_unit.js [hunk 3]
```diff
@@ -1160,6 +1160,71 @@ describe('HlsParser', () => {
...
+          stream.language = 'en';
+          stream.roles = [
+            'public.accessibility.describes-video',
+            'public.accessibility.describes-music-and-sound',
+          ];
+        });
+      });
+      manifest.addPartialTextStream((stream) => {
+        stream.language = 'en';
+        stream.kind = TextStreamKind.SUBTITLE;
```
Comment (joeyparrish) [inline test/hls/hls_parser_unit.js:1209]: This test doesn't cover the new functionality.  The purpose is to add the subtitle role, but this test doesn't set expectations on the role of this in the output.

File: test/hls/hls_parser_unit.js [hunk 4]
```diff
@@ -1160,6 +1160,71 @@ describe('HlsParser', () => {
...
+      });
+      manifest.addPartialTextStream((stream) => {
+        stream.language = 'en';
+        stream.kind = TextStreamKind.SUBTITLE;
+        stream.roles = [
+          'subtitle',
```
Comment (joeyparrish) [inline test/hls/hls_parser_unit.js:1215]: This should be public.accessibility.describes-music-and-sound instead.

File: test/hls/hls_parser_unit.js [hunk 5]
```diff
@@ -1160,6 +1160,71 @@ describe('HlsParser', () => {
...
+      'public.accessibility.describes-music-and-sound",',
+      'NAME="English (describes-video)",URI="audio2"\n',
+
+      '#EXT-X-MEDIA:TYPE=SUBTITLES,GROUP-ID="sub1",LANGUAGE="en",',
+      'NAME="English (caption)",DEFAULT=YES,AUTOSELECT=YES,',
+      'URI="text"\n',
+
+      '#EXT-X-MEDIA:TYPE=SUBTITLES,GROUP-ID="sub1",LANGUAGE="en",',
+      'NAME="English (caption)",DEFAULT=YES,AUTOSELECT=YES,',
+      'public.accessibility.describes-music-and-sound",',
```
Comment (joeyparrish) [inline test/hls/hls_parser_unit.js:1184]: This is malformed and missing CHARACTERISTICS

**Review by joeyparrish — CHANGES_REQUESTED** [review]

File: test/hls/hls_parser_unit.js (hunk 5 above)
Comment (joeyparrish) [reply test/hls/hls_parser_unit.js]: Fixed

File: test/hls/hls_parser_unit.js (hunk 4 above)
Comment (joeyparrish) [reply test/hls/hls_parser_unit.js]: Fixed

File: test/hls/hls_parser_unit.js (hunk 3 above)
Comment (joeyparrish) [reply test/hls/hls_parser_unit.js:1210]: Fixed.  I also set explicit expectations for role in all cases, including the empty array.

**Review by joeyparrish — APPROVED** [review]
Updated code and tests are now passing

**Comment by joeyparrish** [comment]
Thanks for your contribution!

**Comment by officialbidisha** [comment]
Thank you @joeyparrish

**Pull request merged**